### PR TITLE
14166 add l1 fees

### PR DESCRIPTION
**Pull request by endulab**
### What does the PR do

All fees (suggested fees, l1 fee, gas units) are computed in 1 status-go call. 
Suggested fees contains correct l1 fee.
We display the cost with included l1 fee.

Related status-go [pr](https://github.com/status-im/status-go/pull/5071)

### CODE REVIEW

File: src/app_service/service/community_tokens/service.nim [hunk 1]
```diff
@@ -1083,13 +1080,15 @@ QtObject:
     let (ethCurrency, fiatCurrency) = self.create0CurrencyAmounts()
     return ComputeFeeArgs(ethCurrency: ethCurrency, fiatCurrency: fiatCurrency, errorCode: errorCode)
 
+  # Returns eth value with l1 fee included
   proc computeEthValue(self:Service, gasUnits: int, suggestedFees: SuggestedFeesDto): float =
     try:
       let maxFees = suggestedFees.maxFeePerGasM
       let gasPrice = if suggestedFees.eip1559Enabled: maxFees else: suggestedFees.gasPrice
```
Comment (saledjenic) [inline src/app_service/service/community_tokens/service.nim:1087]: This line looks misleading. I mean, how `maxFeePerGasM` can be used as `gasPrice`?

File: src/app_service/service/ens/utils.nim [hunk 2]
```diff
@@ -87,6 +88,15 @@ proc buildTransaction*(
   else:
     result.txType = "0x00"
 
+proc buildTransactionDataDto*(gasUnits: int, suggestedFees: SuggestedFeesDto, addressFrom: string, chainId: int, contractAddress: string): TransactionDataDto =
+    if suggestedFees == nil:
+      error "Can't find suggested fees for chainId", chainId=chainId
+      return
+    return buildTransaction(parseAddress(addressFrom), 0.u256, $gasUnits,
+      if suggestedFees.eip1559Enabled: "" else: $suggestedFees.gasPrice, suggestedFees.eip1559Enabled,
```
Comment (saledjenic) [inline src/app_service/service/ens/utils.nim:96]: @endulab gas price should not depend on eip1559 and should not be empty.

File: src/app_service/service/community_tokens/service.nim (hunk 1 above)
Comment (endulab) [reply src/app_service/service/community_tokens/service.nim:1087]: For EIP-1559 maxFeePerGasM is the absolute maximum you are willing to pay per unit of gas to get your transaction included in a block. You are sure that you will not pay more than that.
So, I understand that you will pay no more than: gas* maxFeePerGasM
For legacy tx you just pay gas*gasPrice.
This is how I understand this.

File: src/app_service/service/ens/utils.nim (hunk 2 above)
Comment (endulab) [reply src/app_service/service/ens/utils.nim:96]: Why gas price is needed for eip1559 tx?

**Comment by saledjenic** [comment]
@endulab you're absolutely right, I overlooked something.

**Review by jrainville — APPROVED** [review]
Looks good. Was there an issue for this or it's just a fix/update to the last one?

Also, is there a status-go PR too?

**Comment by endulab** [comment]
@jrainville what do you mean?
The issue is [here](https://github.com/status-im/status-desktop/issues/14166)
Status-go pr is [here](https://github.com/status-im/status-go/pull/5071)

**Pull request merged**